multiplex_immuno_processing/generate_alignment_paramters.py

```diff
@@ -442,32 +153,89 @@ def os_swap(x):
+
+        #target_img_padded = np.pad(refimg, ((0, 0), (50, 50), (50, 50)), mode='constant')
+        # tifffile.imwrite("refimg.tiff", refimg)
+        alignment_offsets_xyz_list_ORB= []
+
+
+        for i in range(len(img_list)):
+            try:
+                print(f"looking at {i}")
+                new_source_img = img_list[i]
+                source_img_FOV_cropped = new_source_img[:, source_cropping:np.shape(new_source_img)[1]-source_cropping, source_cropping:np.shape(new_source_img)[2]-source_cropping]
+
+                #source_img_padded = np.pad(source_img_FOV, ((0, 0), (50, 50), (50, 50)), mode='constant')
+                print(f"shape of ref img is {np.shape(refimg)}")
+                print(f"shape of source image is {np.shape(source_img_FOV_cropped)}")
+                # tifffile.imwrite(f"source_img_FOV_{i}_padded.tiff", source_img_FOV_cropped)
+
+                #ref_img_cropped = refimg[:,100:np.shape(refimg)[1]-100, 100:np.shape(refimg)[2]-100]
+                # the indexing here is z,y,x -- check this
+                final_y_offset, final_x_offset = registration_utils.perform_alignment(source_img_FOV_cropped, refimg, smaller_fov_modality="source", scale_factor_xy=1, scale_factor_z=1, source_alignment_channel=0, target_alignment_channel=0, source_output_channel=[0], target_output_channel=[0], prealign_z=True, denoise_z=True, use_refinement=False, save_composite=False)
+                print(final_y_offset, final_x_offset)
+                # match_list = align_helper.return_aligned_img_list_new(img_list,offset_list)
+                print("for first img in new_source_img, x_offset is", final_x_offset)
+                print("for first img in new_source_img, y_offset is", final_y_offset)
+                alignment_offsets_xyz_list_ORB.append((final_x_offset, final_y_offset))
+            except:
+                alignment_offsets_xyz_list_ORB.append((0,0))
+
+        final_version_offsets_zyx=[]
+        # This just copies over the z offset from the cross correlation method
+        for k in range(len(alignment_offsets_xyz_list_ORB)):
+            dat=alignment_offsets_xyz_list_ORB[k]
+            final_version_offsets_zyx.append([alignment_offsets_xyz_list[k][0], source_cropping-dat[0], source_cropping-dat[1]])
+        
+        print(f"final version offset xyz is {final_version_offsets_zyx}")
+        print("outputing")
+        dfimg["alignment_offsets_xyz_ORB"] = final_version_offsets_zyx
+        dfimg["alignment_offsets_xyz_cross_corr"] = alignment_offsets_xyz_list
+
+    elif args.method=="cross_cor":
+        dfimg["alignment_offsets_xyz_cross_corr"] = alignment_offsets_xyz_list
+        dfimg["alignment_offsets_xyz_ORB"] = []
+        
+
+
+    print("dfimg.shape[0] is {}".format(dfimg.shape[0]))
+    dfimg["template_position"] = [Position] * dfimg.shape[0] # This is what?
+    dfout_p = dfimg[["template_position", "align_channel", "alignment_offsets_xyz_ORB", "alignment_offsets_xyz_cross_corr"]]
```

Comment:
there should be a column named "method" that tells which method was used rather than changing the name of `alignment_offsets_xyz`. (i should note that one needed change in a future update is to rename `alignment_offsets_xyz` to `alignment_offsets_zyx` since that is the actual order of the offsets).

Reply:
Will come back to this